lib/p2p/NodeList.ts

```diff
@@ -69,13 +69,36 @@ class NodeList {
   public updateAddresses = async (nodePubKey: string, addresses: Address[] = []) => {
     const node = this.nodes.get(nodePubKey);
     if (node) {
-      node.addresses = addresses;
+      // avoid overriding the `lastConnected` field for existing matching addresses unless a new value was set
+      node.addresses = addresses.map((newAddress) => {
+        const oldAddress = node.addresses.find(address => address.host === newAddress.host && address.port === newAddress.port);
```

Comment:
Might be nice to have an equality method in `addressUtils` for this logic since we're using in a couple of places.